packages/react/src/modules/slack/components/ConnectedSlackChannelPicker/ConnectedSlackChannelPicker.tsx

```diff
@@ -0,0 +1,216 @@
+  const [isSettingChannels, setIsSettingChannels] = useState(false);
+  const [showChannelPicker, setShowChannelPicker] = useState(false);
+  const [hasError, setHasError] = useState(false);
+
+  const { data, isLoading: isLoadingSlackChannels } = useSlackChannels(
+    tenant,
+    connectionsObject,
+    host,
+    knockClientId,
+    knockSlackChannelId,
+    userToken,
+    user,
+    toggleForRefetch,
+    setHasError,
+    hasError,
+  );
+
+  const isLoading = isSettingChannels || isLoadingSlackChannels;
+
+  const { channels } = data;
+
+  const handleOptionClick = async (channelId: string) => {
+    setIsSettingChannels(true);
+    const channelToUpdate = channels.find(
+      (channel: Channel) => channel.id === channelId,
+    );
+    if (channelToUpdate) {
+      const updatedChannels = channels.map((channel: Channel) =>
+        channel.id === channelId
+          ? { ...channel, connected: !channel.connected }
+          : channel,
+      );
+
+      const channelsToSendToKnock = updatedChannels
+        .filter((channel: Channel) => channel.connected)
+        .map((channel: Channel) => {
+          return {
+            channel_id: channel.id,
+          };
+        });
+
+      try {
+        await knockClient.slack.setChannelConnections({
+          objectId: connectionsObject.objectId,
+          collection: connectionsObject.collection,
+          knockChannelId: knockSlackChannelId,
+          slackChannelId: channelId,
+          connections: channelsToSendToKnock,
+          userId: user.id,
+        });
+        setToggleForRefetch((state) => !state);
+      } catch (error) {
+        setHasError(true);
+        setShowChannelPicker(false);
+      }
+      setIsSettingChannels(false);
+    }
+  };
+
+  const channelsConnectedMessage = () => {
+    if (hasError) {
+      return "Error fetching channels";
+    }
+
+    if (isLoading) {
+      return "Loading...";
+    }
+
+    const numberConnectedChannels = channels.filter(
+      (channel: Channel) => channel.connected,
+    ).length;
+
+    if (numberConnectedChannels === 0) {
+      return "Select a Slack channel";
+    }
+
+    if (numberConnectedChannels === 1) {
+      const connectedChannel = channels.find(
+        (channel: Channel) => channel.connected,
+      ) as Channel;
+      return `#${connectedChannel.name}`;
+    }
+
+    return "Multiple channels connected";
+  };
+  return (
+    <div
+      style={{
+        width: "256px",
+      }}
+    >
+      <div>
+        <div className="multiSelectAccordion">
+          <div
+            key={"select"}
+            className="header"
+            onClick={() => {
+              if (!hasError) {
+                setShowChannelPicker(!showChannelPicker);
+              }
+            }}
+          >
+            {channelsConnectedMessage()}
+            {!hasError && (
+              <span className="icon">
+                {showChannelPicker ? (
+                  <svg
```

Comment:
Nit but it'd be nice to break out these icons into separate components (could have an icons folder) as it'd make this component a bit more concise.

Reply:
okay i think i'll tackle this in another PR as we're probably going to change a lot of the look of this, then will introduce that